### ISSUE TITLE
Show hint on focus is not working

### ISSUE THREAD

**Issue by iamzozo** [bassjobsen/Bootstrap-3-Typeahead#192]
On init, the lookup method always hide the dropdown:

```javascript
if (this.query.length < this.options.minLength) {
  return this.shown ? this.hide() : this;
}
```

Also I think the click event should hide the dropdown, when a value is selected.

**Comment by bassjobsen** [comment]
> Also I think the click event should hide the dropdown, when a value is selected.
Your PR also remove the focus. But shouldn't the click focus the element?

**Comment by iamzozo** [comment]
You are right, it should. But right now if `showHintOnFocus` is active, set focus will show the dropdown again. Using a state and state check could solve the problem, for example "closing"?

**Comment by bassjobsen** [comment]
Okay, than i think we should have this state check. It is not in your PR right now? Thank for your contributions in the first place!

**Comment by HeppyKarlsson** [comment]
I just tried to set showHintOnFocus and minLength and now it shows on focus. 
`{
showHintOnFocus: true,
minLength: 0
}`

**Comment by iamzozo** [comment]
Thx, I will check it out, if it works, I'm going to remove the request, or update the code as we discussed.

**Comment by HeppyKarlsson** [comment]
Although now I'm getting the problem that when selecting a option it opens the "hints" again as discussed above.
